src/wallet/wallet.cpp

```diff
@@ -1336,13 +1313,27 @@ void CWallet::MarkConflicted(const uint256& hashBlock, int conflicting_height, c
     if (conflictconfirms >= 0)
         return;
 
-    // Do not flush the wallet here for performance reasons
+    auto try_updating_state = [&](CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) {
+        if (conflictconfirms < GetTxDepthInMainChain(wtx)) {
+            wtx.m_state = TxStateConflicted{hashBlock, conflicting_height};
```

Comment:
In commit "wallet: introduce generic recursive tx state updating function" (f998cdbc689cc65ff0aca98ce63758c59786c1db)

Would suggest keeping the previous comment here instead of deleting it. It's not obvious what the check means or what's happening in the block:

```c++
// Block is 'more conflicted' than current confirm; update.
// Mark transaction as conflicted with this block.
```

Reply:
Done